### ISSUE TITLE
[🐛BUG] 对ration_split划分方式计算指标疑惑

### ISSUE THREAD

**Issue by ItsukiFujii** [RUCAIBox/RecBole#1444]
RecBole框架在测试指标时，在leave_one_out策略下没有问题，但是ratio_split策略下时，将每个用户的行为作为单条记录，这种做法在计算recall，ndcg这种需要聚合用户行为的指标时，貌似不合理。

**Comment by zhengbw0324** [comment]
@ItsukiFujii 
ratio_split策略下，如果模型需要聚合用户行为时，会使用训练集中的用户交互历史。

**Comment by ItsukiFujii** [comment]
@zhengbw0324 我在跑Amazon-beauty 5core数据集（user_num: 22363, item_num: 12101）采用RS策略8:1:1时，观察了evaluators.metrics.Recall这个类的数据流动
![image](https://user-images.githubusercontent.com/42373615/191767488-47c55edd-40e0-42ea-885d-f539bf3d8724.png)
这个result的第0维度并不是22363，而是2.4w+，同一用户的记录被作为了当都一条

**Comment by zhengbw0324** [comment]
@ItsukiFujii 
维度是 2.4w+ 而不是 22363 的原因是在 Amazon-beauty 5core 数据集中，交互记录少于5个的用户被移除了，因此刚开始时数据信息显示用户数量为 22363。实际上所有用户数量为2.4w+。

**Comment by ItsukiFujii** [comment]
## 这是我采用的配置
General Hyper Parameters:
gpu_id = 0
use_gpu = True
seed = 2022
state = INFO
reproducibility = True
data_path = dataset/beauty
checkpoint_dir = saved
show_progress = True
save_dataset = False
dataset_save_path = None
save_dataloaders = False
dataloaders_save_path = None
log_wandb = False

Training Hyper Parameters:
epochs = 300
train_batch_size = 2048
learner = adam
learning_rate = 0.001
neg_sampling = None
eval_step = 1
stopping_step = 10
clip_grad_norm = None
weight_decay = 0.0
loss_decimal_place = 4

Evaluation Hyper Parameters:
eval_args = {'split': {'RS': [0.8, 0.1, 0.1]}, 'order': 'TO', 'mode': 'full', 'group_by': 'user'}
repeatable = True
metrics = ['Recall', 'NDCG', 'Hit']
topk = [5, 10, 20]
valid_metric = NDCG@10
valid_metric_bigger = True
eval_batch_size = 4096
metric_decimal_place = 4

Dataset Hyper Parameters:
field_separator = 	
seq_separator =  
USER_ID_FIELD = user_id
ITEM_ID_FIELD = item_id
RATING_FIELD = rating
TIME_FIELD = timestamp
seq_len = None
LABEL_FIELD = label
threshold = None
NEG_PREFIX = neg_
load_col = {'inter': ['user_id', 'item_id', 'rating', 'timestamp']}
unload_col = None
unused_col = None
additional_feat_suffix = None
rm_dup_inter = None
val_interval = None
filter_inter_by_user_or_item = True
user_inter_num_interval = None
item_inter_num_interval = None
alias_of_user_id = None
alias_of_item_id = None
alias_of_entity_id = None
alias_of_relation_id = None
preload_weight = None
normalize_field = None
normalize_all = True
ITEM_LIST_LENGTH_FIELD = item_length
LIST_SUFFIX = _list
MAX_ITEM_LIST_LENGTH = 50
POSITION_FIELD = position_id
HEAD_ENTITY_ID_FIELD = head_id
TAIL_ENTITY_ID_FIELD = tail_id
RELATION_ID_FIELD = relation_id
ENTITY_ID_FIELD = entity_id
benchmark_filename = None

Other Hyper Parameters: 
wandb_project = recbole
require_pow = False
embedding_size = 64
hidden_size = 128
num_layers = 1
dropout_prob = 0.3
loss_type = CE
MODEL_TYPE = ModelType.SEQUENTIAL
MODEL_INPUT_TYPE = InputType.POINTWISE
eval_type = EvaluatorType.RANKING
device = cuda
train_neg_sample_args = {'strategy': 'none'}
eval_neg_sample_args = {'strategy': 'full', 'distribution': 'uniform'}


Fri 23 Sep 2022 09:02:33 INFO  beauty
The number of users: 22364
Average actions of users: 8.876358270357287
The number of items: 12102
Average actions of items: 16.403768283612923
The number of inters: 198502
The sparsity of the dataset: 99.92665707018277%
Remain Fields: ['user_id', 'item_id', 'rating', 'timestamp']

### 模型时gru4rec是序列推荐模型，划分策略是RS_TO

![image](https://user-images.githubusercontent.com/42373615/191875701-81ab0a43-9863-48e6-82c4-d0d7c2d51887.png)
result的第0维度是24683

**Comment by zhengbw0324** [comment]
@ItsukiFujii 
你可以看一下验证集的用户数量，应该是24683，这个数据集比较稀疏，有的用户不会出现在验证集里。

**Comment by zhengbw0324** [comment]
@ItsukiFujii 
可以在 `quick_start.py` 的第73行加一句 `print(len(valid_data.uid_list))` 看看输出。

**Comment by ItsukiFujii** [comment]
@zhengbw0324 
![image](https://user-images.githubusercontent.com/42373615/191879166-4f6396e5-51ec-4189-9545-d2c786d9dcb9.png)
从上至下是train, valid, test数据集，实际上都是22364
但是我现在大概明白了，序列推荐模型为什么计算指标不按用户维度聚合了，感谢！！！
